File: app/src/main/java/com/perflyst/twire/views/recyclerviews/auto_span_behaviours/StreamAutoSpanBehaviour.java
package com.perflyst.twire.views.recyclerviews.auto_span_behaviours;

import android.content.Context;

import com.perflyst.twire.R;
import com.perflyst.twire.service.Settings;

/**
 * Created by Sebastian Rask on 09-05-2017.
 */

public class StreamAutoSpanBehaviour implements AutoSpanBehaviour {
    @Override
    public String getElementSizeName() {
        return Settings.getAppearanceStreamSize();
    }

    @Override
    public int getElementWidth(Context context) {
        return (int) context.getResources().getDimension(R.dimen.stream_card_min_width) + (int) context.getResources().getDimension(R.dimen.stream_card_left_margin);
    }
}

File: app/src/main/java/com/perflyst/twire/views/recyclerviews/auto_span_behaviours/StreamAutoSpanBehaviour.kt
package com.perflyst.twire.views.recyclerviews.auto_span_behaviours

import android.content.Context
import com.perflyst.twire.R
import com.perflyst.twire.service.Settings.appearanceStreamSize

/**
 * Created by Sebastian Rask on 09-05-2017.
 */
class StreamAutoSpanBehaviour : AutoSpanBehaviour {
    override val elementSizeName: String get() = appearanceStreamSize

    override fun getElementWidth(context: Context): Int {
        return context.resources.getDimension(R.dimen.stream_card_min_width)
            .toInt() + context.resources.getDimension(R.dimen.stream_card_left_margin).toInt()
    }
}
